src/effects/RealtimeEffectList.cpp

```diff
@@ -85,14 +85,20 @@ void RealtimeEffectList::Visit(StateVisitor func)
 RealtimeEffectState *RealtimeEffectList::AddState(const PluginID &id)
 {
    auto pState = std::make_unique<RealtimeEffectState>(id);
-   if (id.empty() || pState->GetEffect() != nullptr) {
+   if (pState->GetEffect() != nullptr) {
```

Comment:
You lost the conjunct `id.empty()` but it is important not to return nullptr in that case.  It will happen when reloading a `RealtimeEffectList` from a file.  The state is constructed first, then the id is read from the file and assigned to it.